django/db/models/fields/related.py

```diff
@@ -26,74 +26,58 @@
 RECURSIVE_RELATIONSHIP_CONSTANT = 'self'
 
 
-def add_lazy_relation(cls, field, relation, operation):
+def resolve_relation(scope_model, relation, always_text=False):
```

Comment:
In general, I find boolean flag parameters that fundamentally change the operation of a function to be an API smell. (Not sure if there's a name for this principle, but it's often referred to in discussions of Python stdlib APIs.) As an alternative, I think it would be cleaner to have a separate utility function that takes either a string or a model, and returns a string, and then in the place where you'd use the `always_text` flag, wrap the call to `resolve_relation` with a call to that utility.

(Yes, that's an additional function call, but we first optimize for readability and nice APIs, and then optimize for performance when we have data showing where our bottleneck is.)